### ISSUE TITLE
Benchmarking a new load balancing algorithm

### ISSUE THREAD

**Issue by Dim131** [smallrye/smallrye-stork#603]
TLDR: We would like to benchmark a simple variant of the Two-Choice load balancing algorithm on smallrye-stork. Is there a recommended benchmark or a typical workload that you use?

The variant is the following: For each request,
 1. Allocate to the least loaded of two randomly chosen servers (i.e., like Two-Choice) with probability β.
 2. Otherwise, allocate to a random server.

Somewhat surprisingly we recently demonstrated in theory and simulations ([paper](https://arxiv.org/pdf/2302.04399.pdf), [slides](https://www.cl.cam.ac.uk/~dl516/slides/spaa23_slides.pdf), [visualisation](https://dimitrioslos.com/phdthesis/conf/spaa23.html)) that for suitably chosen β, this outperforms Two-Choice when there is outdated information (RTT is large, there are multiple allocators) or there is noise. The main reason for this is that it allocates to lesser loaded servers less aggressively than Two-Choice, so that it spreads the load more evenly (see figure below).

![spaa23_tc_vs_opb](https://github.com/smallrye/smallrye-stork/assets/9382526/79b9131a-4fb1-447b-99ca-a8415fdb99f3)

We would like to see whether this result also implies improvements in real-world load balancers, including smallrye-stork. For the evaluation purposes, I have drafted a very simple implementation (see [here](https://github.com/Dim131/smallrye-stork)) and was wondering whether you could give me some pointers for setting up and running a benchmark representative of a typical workload.

Kind regards, 
Dimitris 

P.S. I have reached out to a few other open source load balancers.

**Comment by cescoffier** [comment]
That's interesting! 

I would start with micro-benchmarks, just stressing the service discovery.
You can use JMH, but it might be a bit overkill. That being said, what will give you the best "results"

**Comment by Dim131** [comment]
Thank you for the reply! I will start implementing a benchmark and will try to use JMH. But I have a few questions first:  
 1. How many instances of stork do you expect to be running in parallel? Is it one instance on a single core? Multiple instances on multiple cores? Or multiple instances multiple machines?
 2. What is the typical range for the number of instances running? Are these expected to be on the same or different machines?
 3. When you say "just stressing the servise discovery" which parameters for the above did you have in mind?
